spec/record_spec.rb

```diff
@@ -70,6 +71,26 @@
     end
   end
 
+  describe "for hash with ActiveRecord::Type::Value type instance" do
+    let(:klass) do
+      Class.new(ActiveRecord::Base) do
+        include AttrJson::Record
+
+        self.table_name = "products"
+        attr_json :value, ActiveRecord::Type::Value.new
+      end
+    end
+
+    it "stores the hash value directly without extra escaping" do
+      value = {"a" => {"b" => "c"}}
+      instance.value = value
+      instance.save
+      #load the instance directly from the database
+      instance_loaded_raw = JSON.parse(ActiveRecord::Base.connection.execute(instance.class.all.to_sql).first["json_attributes"])
+      expect(instance_loaded_raw["value"]).to eq(value)
```

Comment:
Huh, if it's really raw... how the heck is it coming back as an actual Hash rather than a serialized hash? I'd actually expect it as a String serialized hash here! (Just not, like a... double-serialized hash?) But something in Rails is still reifying it into a Hash before we get it?  Escaping is confusing.

Reply:
Since it is backed by Postgres `jsonb` I think it is expected that it would be converted to an arbitrary-depth Ruby hash, right? After all, `json` is analogous to a combination of Hashes and Arrays in Ruby-land, isn't it?

Reply:
Well, _postgres_ can't convert it to an arbitrary-depth ruby-hash. (Technically any JSON data, the first object can be an array, or a primitive, too).  Postgres doens't know ruby hashes. 

ActiveRecord does though, yes, since all of the attr_json stuff is stored in an ActiveRecord json/jsonb column. 

But this particular test is testing a method of access that skips that ActiveRecord conversion. So it really does come back as a big serialized string. 

But then the `JSON.parse` on line 89 (which I had missed in my comment), turns it back into ruby data, ensuring that it is valid JSON and is what we expect.

Reply:
I see, good to know! 😀🔍